openpype/hooks/pre_copy_last_published_workfile.py

```diff
@@ -0,0 +1,199 @@
+
+        # Check it can proceed
+        if not project_doc and not asset_doc:
+            return
+
+        # Get subset id
+        subset_id = next(
+            (
+                subset["_id"]
+                for subset in get_subsets(
+                    project_name,
+                    asset_ids=[asset_doc["_id"]],
+                    fields=["_id", "data.family", "data.families"],
+                )
+                if subset["data"].get("family") == "workfile"
+                # Legacy compatibility
+                or "workfile" in subset["data"].get("families", {})
+            ),
+            None,
+        )
+        if not subset_id:
+            self.log.debug('No any workfile for asset "{}".').format(
+                asset_doc["name"]
+            )
+            return
+
+        # Get workfile representation
+        workfile_representation = next(
+            (
+                representation
+                for representation in get_representations(
+                    project_name,
+                    version_ids=[
+                        (
+                            get_last_version_by_subset_id(
+                                project_name, subset_id, fields=["_id"]
+                            )
+                            or {}
+                        ).get("_id")
+                    ],
+                )
+                if representation["context"]["task"]["name"] == task_name
+            ),
+            None,
+        )
+
+        if not workfile_representation:
+            self.log.debug(
+                'No published workfile for task "{}" and host "{}".'
+            ).format(task_name, host_name)
+            return
+
+        # POST to webserver sites to add to representations
+        webserver_url = os.environ.get("OPENPYPE_WEBSERVER_URL")
+        if not webserver_url:
+            self.log.warning("Couldn't find webserver url")
+            return
+
+        entry_point_url = "{}/sync_server".format(webserver_url)
+        rest_api_url = "{}/add_sites_to_representations".format(
+            entry_point_url
+        )
+        try:
+            import requests
+        except Exception:
+            self.log.warning(
+                "Couldn't add sites to representations "
+                "('requests' is not available)"
+            )
+            return
+
+        local_site_id = get_local_site_id()
+        requests.post(
+            rest_api_url,
+            json={
+                "project_name": project_name,
+                "sites": [local_site_id],
+                "representations": [str(workfile_representation["_id"])],
+            },
+        )
+
+        # Wait for the download loop to end
+        last_created_time = get_representation_last_created_time_on_site(
+            workfile_representation, local_site_id
+        )
+        while (
+            last_created_time
+            >= get_representation_last_created_time_on_site(
```

Comment:
I would actually expect that `requests.get(...)` is used here instead.

Reply:
Come on, I changed it on the request from @kalisp...